### PR TITLE
Rename notifications-backend module folder

### PR DESCRIPTION
**Pull request by gwenneg**
The path to the `notifications-backend` module should be `/notifications/backend`. This PR renames the folder.

### CODE REVIEW

File: backend/src/main/docker/Dockerfile-build.jvm [hunk 1]
```diff
@@ -5,10 +5,10 @@
...
 FROM registry.access.redhat.com/ubi8/openjdk-11:latest AS build
 USER root
-COPY src /home/jboss/src
-COPY pom.xml /home/jboss
-COPY checkstyle.xml /home/jboss
-COPY checkstyle-suppressions.xml /home/jboss
+COPY backend/src /home/jboss/src
+COPY backend/pom.xml /home/jboss
+COPY backend/checkstyle.xml /home/jboss
+COPY backend/checkstyle-suppressions.xml /home/jboss
```
[the remark this answers is not in the document]
Comment (josejulio) [reply backend/src/main/docker/Dockerfile-build.jvm:11]: Doesn't the parent module has any shared setting that is needed by this one?

[the remark this answers is not in the document]
Comment (gwenneg) [reply backend/src/main/docker/Dockerfile-build.jvm:11]: You're right, the build won't succeed without the parent pom.

[the remark this answers is not in the document]
Comment (gwenneg) [reply backend/src/main/docker/Dockerfile-build.jvm]: @josejulio It should be better now.

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**